hive: TestBroadcastPeers flakes

```
2020-07-27T13:40:12.1434860Z === RUN   TestBroadcastPeers
2020-07-27T13:40:12.1438930Z === RUN   TestBroadcastPeers/OK_-_single_record
2020-07-27T13:40:12.1442320Z === RUN   TestBroadcastPeers/OK_-_single_batch_-_multiple_records
2020-07-27T13:40:12.1445390Z === RUN   TestBroadcastPeers/OK_-_single_batch_-_max_number_of_records
2020-07-27T13:40:12.1484820Z ##[error]    hive_test.go:207: Overlays got [ec4c18ae1ebec7b5f5f7e88e8892f118ab9966b001067d0e5a2727f17d56cf7c d23f762b7e0d7980bb2b21d78c45000a6b1370f6d8e65da7ac52df1ae10196fd 7de8f721022c9bc760d2ffdae1235b0a81ece4c6fd8774712256ae29385e0246 6e512e1fbc58d4e0e1a0f68e44585fb31d96b30dafeb12d3eb1f4e177cf68dc5 91a46a8c49746752e9848ea431805037e22827d26b2621f5a5c1c15037fff343 6974d8ed4ba7b475a133425fd992667573567251b7470a30d1ea0e9f4d1b418c 4a0b95ce097da0cbd4dac3e6ae4327aa18292bfa22f9bcf6b8812322a97c4440 7aa2f9622b4a53ee07ba57be5d99ee2617a4ae9589db1cac33aa44e9caf40650 2434962363fcc5d963d8101a1f66b0c68a0fb39616930c04dd930438df6e2768 b7bedf47433bb72e8744c0eafab99ff1740a25b294a2665a75ce48543f13b037 9b004c1b7c1a0dbc33c4bf74aa98ead9b3bd7d33213b5ca80d003cb451c09317 43ce1f9064c35ff8dbdc7d45f8338687456121364c96d40951f2854be7a09aa3 e7be859d695bdf6a19049ffb13eb1d1d2f716c742505312063ef2b2e7258b4fe 166964d405a44fef98dc008c5314e1a6628e50028382552c9c7646bbd126862a a7e16af3f2a870ad3f48385c1c6bee34eb06760c7cdde272eed697e32621a196 9cb7355e8b74dfb0804e72460ecebbac4931fd308f08a2ed7a9bcff2ac5768ed 8744fff4f37f8634cf8c3ced8b58a2081d6b7efa96216fbb4b9b30c87ca1fc4a 30d672beaffeb6abb6b4cf6de623a9b8a9b4475847e599837320adea237c9752 b2f74d8295b88a4db51f309865b9caa060c03d0debe7f549693694414344adeb 59f8476fed8e936e752028601fc4e64b23825fd2ab0877c282ab085c75e1896d f73b44c66b4906874ee1fec7e6709d88e026814e75236ad877b385c81351a33c b47e53f415e170ce497108ee06f52f73139a29385e6172f034b3603e84ca3c1d], want [43ce1f9064c35ff8dbdc7d45f8338687456121364c96d40951f2854be7a09aa3 91a46a8c49746752e9848ea431805037e22827d26b2621f5a5c1c15037fff343 8744fff4f37f8634cf8c3ced8b58a2081d6b7efa96216fbb4b9b30c87ca1fc4a 6974d8ed4ba7b475a133425fd992667573567251b7470a30d1ea0e9f4d1b418c 4a0b95ce097da0cbd4dac3e6ae4327aa18292bfa22f9bcf6b8812322a97c4440 166964d405a44fef98dc008c5314e1a6628e50028382552c9c7646bbd126862a 7aa2f9622b4a53ee07ba57be5d99ee2617a4ae9589db1cac33aa44e9caf40650 b47e53f415e170ce497108ee06f52f73139a29385e6172f034b3603e84ca3c1d 6e512e1fbc58d4e0e1a0f68e44585fb31d96b30dafeb12d3eb1f4e177cf68dc5 2434962363fcc5d963d8101a1f66b0c68a0fb39616930c04dd930438df6e2768 30d672beaffeb6abb6b4cf6de623a9b8a9b4475847e599837320adea237c9752 a7e16af3f2a870ad3f48385c1c6bee34eb06760c7cdde272eed697e32621a196 b2f74d8295b88a4db51f309865b9caa060c03d0debe7f549693694414344adeb ec4c18ae1ebec7b5f5f7e88e8892f118ab9966b001067d0e5a2727f17d56cf7c 59f8476fed8e936e752028601fc4e64b23825fd2ab0877c282ab085c75e1896d b7bedf47433bb72e8744c0eafab99ff1740a25b294a2665a75ce48543f13b037 9b004c1b7c1a0dbc33c4bf74aa98ead9b3bd7d33213b5ca80d003cb451c09317 f73b44c66b4906874ee1fec7e6709d88e026814e75236ad877b385c81351a33c d23f762b7e0d7980bb2b21d78c45000a6b1370f6d8e65da7ac52df1ae10196fd e7be859d695bdf6a19049ffb13eb1d1d2f716c742505312063ef2b2e7258b4fe 9cb7355e8b74dfb0804e72460ecebbac4931fd308f08a2ed7a9bcff2ac5768ed 7de8f721022c9bc760d2ffdae1235b0a81ece4c6fd8774712256ae29385e0246 27f8412e061a3025c472f1543acb7ce783df7472a5cfed4d079cea7a98d5bbd0 4e42fc9674517b6d3635e36d513b526cea4bbde6cfdb67616c1315b4a83286b3 07d1109c95ceb6361a6100d278e7901c258a3172f931626ec99823cf5a14927d 4640b022ee4692d2880e8f0e228ba1c0fde2015118f2cab40c004095f92410fc d14c184434948770a39005b764b8e57d78cfa76369fcd2debb9c66190c26477b 45336dfeac41982b0ce4bb72892420b3e4d9ab2ed7b4a00b66b2db6abf4159a8 612e27f8561fb6d48e72b2acdbeff5671f9ee1b7e20a133eac8bd218d27bf472 5b7c8b7b044068e1cc8f0a774e9772f9b31deb8c3536f49e70a5e5d6fc4e0a04]
2020-07-27T13:40:12.1501680Z === RUN   TestBroadcastPeers/OK_-_multiple_batches
2020-07-27T13:40:12.1503610Z === RUN   TestBroadcastPeers/OK_-_multiple_batches_-_max_number_of_records
2020-07-27T13:40:12.1525980Z ##[error]    hive_test.go:207: Overlays got [6e512e1fbc58d4e0e1a0f68e44585fb31d96b30dafeb12d3eb1f4e177cf68dc5 4bbe8eda2a48dbcd450ffa278fc1694cf712760a764bd78261e7147ef81bbdc9 a7e16af3f2a870ad3f48385c1c6bee34eb06760c7cdde272eed697e32621a196 43ce1f9064c35ff8dbdc7d45f8338687456121364c96d40951f2854be7a09aa3 d67db4b0f8100a46eb0600e92aff2bca4fefaaaeeda41fab18d35b506ef8de51 d23f762b7e0d7980bb2b21d78c45000a6b1370f6d8e65da7ac52df1ae10196fd d14c184434948770a39005b764b8e57d78cfa76369fcd2debb9c66190c26477b 6974d8ed4ba7b475a133425fd992667573567251b7470a30d1ea0e9f4d1b418c b7bedf47433bb72e8744c0eafab99ff1740a25b294a2665a75ce48543f13b037 9b004c1b7c1a0dbc33c4bf74aa98ead9b3bd7d33213b5ca80d003cb451c09317 4a0b95ce097da0cbd4dac3e6ae4327aa18292bfa22f9bcf6b8812322a97c4440 8f63380315c23754ccccb6034afdc818a22978c0cd294587a55d0c497a839e18 93b89806b7a85921780f5dd2a28bb3cc9f8fc273e23d699a82c8fc6251a5d4ca b47e53f415e170ce497108ee06f52f73139a29385e6172f034b3603e84ca3c1d f73b44c66b4906874ee1fec7e6709d88e026814e75236ad877b385c81351a33c 9cb7355e8b74dfb0804e72460ecebbac4931fd308f08a2ed7a9bcff2ac5768ed 5f8fd3228d034715f202c9eac40f03406ca93d97eaac09a5b72207ddd2419439 f17c40af1f09a99bc6138da9fa75655267c1f22ca4b3f4ec2d03d36ee118484d 9cbfc8093367fb9ff6089283daa3b53c98980a5ce2ac88f514d090ccb4a54da6 ec4c18ae1ebec7b5f5f7e88e8892f118ab9966b001067d0e5a2727f17d56cf7c 45336dfeac41982b0ce4bb72892420b3e4d9ab2ed7b4a00b66b2db6abf4159a8 06ac8783300828ec149a22f6a333f1edf72799e0ddb6568ca94c86a85e84fcef 07d1109c95ceb6361a6100d278e7901c258a3172f931626ec99823cf5a14927d c640f1f5d9130a711ec3e349bd608894ff76c341730c6c7d782077d7c98bddcf 62c879e796c92462ea2de5f164961baf41f6d222db9024495a5fe05c689e4348 bf93ebc964b94dd39b6a14a27bcb1224ff0fb1ce335cc441c494cedc78875dd7 4e42fc9674517b6d3635e36d513b526cea4bbde6cfdb67616c1315b4a83286b3 4640b022ee4692d2880e8f0e228ba1c0fde2015118f2cab40c004095f92410fc 2434962363fcc5d963d8101a1f66b0c68a0fb39616930c04dd930438df6e2768 b2f74d8295b88a4db51f309865b9caa060c03d0debe7f549693694414344adeb ef6c2c6e815edfcdac7f8a05e42eeb71412969358ad89d4d939e7a144bf0f99b 721f7b2a84e1a4d1fe4a395fd1af2827793bab0beff766ae40cbf3a4cb6b9cf9 cb2ccfe2110c9451827b9ac4d87f0d5273422efeeaf84e727a4ead8fbd63b04a 09e5e7e3f1a9f9d3cae97926ed58391f9ad577c168aecfd9ddb5acf6a9110a10 c704a8185ad3cb7e97aab12c5bff71b9f5cbaeb938cb8e2adb5ac0efc946731e 8744fff4f37f8634cf8c3ced8b58a2081d6b7efa96216fbb4b9b30c87ca1fc4a 06e080ba8c0051082f93b69a2b9dfa95f56009f74794a506e95bccf14f2303f2 5d65ad0e3bcfdc6f7bec5c2b07218e414ce1a72919a9095d04bbe7a4eccb4bad 613e7e143552e22269da9326e4256fa4dff44fc1a6688aae7666d066248eb449 0f2da5903dfcf31d7c4421fe9dd910c94800b4401ae791b534b740b81e9fbcc6 22ce2fb606af969b02873ed968f10fb69f105fde801846806159acb0f4bfad2e 7de8f721022c9bc760d2ffdae1235b0a81ece4c6fd8774712256ae29385e0246 d616c20aacb0b9310d3dacc00b285b5224268abd4e8ee37b827364bab65a8f8c 7aa2f9622b4a53ee07ba57be5d99ee2617a4ae9589db1cac33aa44e9caf40650 44e983dd6ee83acd0f744e5a9a5618b846c5bfe81891ac58aaf6fba78c1db8cb c18c93713d2308c8914e9caf3ecd15b3b19090b59960bc5b5ec982a3ee5457c2 dd8bf31ace2648b5cb02803fe6cd112371fa0e22387fa507f07ce22d282e8b11 166964d405a44fef98dc008c5314e1a6628e50028382552c9c7646bbd126862a e7be859d695bdf6a19049ffb13eb1d1d2f716c742505312063ef2b2e7258b4fe 91a46a8c49746752e9848ea431805037e22827d26b2621f5a5c1c15037fff343 925924cbf2d216b85d6c7b3a563b718a3b67293fc7d5a3a09f57659bc15de540 f3efffe4579ab827e5fe2f0f0dc2f8783490dfb305478186f487bb452951d6f1 30d672beaffeb6abb6b4cf6de623a9b8a9b4475847e599837320adea237c9752 59f8476fed8e936e752028601fc4e64b23825fd2ab0877c282ab085c75e1896d 27f8412e061a3025c472f1543acb7ce783df7472a5cfed4d079cea7a98d5bbd0 aeddffd635950418441b864a25ff194fd04c0bc2b7dd9c93c1325afa296f5ae7], want [43ce1f9064c35ff8dbdc7d45f8338687456121364c96d40951f2854be7a09aa3 91a46a8c49746752e9848ea431805037e22827d26b2621f5a5c1c15037fff343 8744fff4f37f8634cf8c3ced8b58a2081d6b7efa96216fbb4b9b30c87ca1fc4a 6974d8ed4ba7b475a133425fd992667573567251b7470a30d1ea0e9f4d1b418c 4a0b95ce097da0cbd4dac3e6ae4327aa18292bfa22f9bcf6b8812322a97c4440 166964d405a44fef98dc008c5314e1a6628e50028382552c9c7646bbd126862a 7aa2f9622b4a53ee07ba57be5d99ee2617a4ae9589db1cac33aa44e9caf40650 b47e53f415e170ce497108ee06f52f73139a29385e6172f034b3603e84ca3c1d 6e512e1fbc58d4e0e1a0f68e44585fb31d96b30dafeb12d3eb1f4e177cf68dc5 2434962363fcc5d963d8101a1f66b0c68a0fb39616930c04dd930438df6e2768 30d672beaffeb6abb6b4cf6de623a9b8a9b4475847e599837320adea237c9752 a7e16af3f2a870ad3f48385c1c6bee34eb06760c7cdde272eed697e32621a196 b2f74d8295b88a4db51f309865b9caa060c03d0debe7f549693694414344adeb ec4c18ae1ebec7b5f5f7e88e8892f118ab9966b001067d0e5a2727f17d56cf7c 59f8476fed8e936e752028601fc4e64b23825fd2ab0877c282ab085c75e1896d b7bedf47433bb72e8744c0eafab99ff1740a25b294a2665a75ce48543f13b037 9b004c1b7c1a0dbc33c4bf74aa98ead9b3bd7d33213b5ca80d003cb451c09317 f73b44c66b4906874ee1fec7e6709d88e026814e75236ad877b385c81351a33c d23f762b7e0d7980bb2b21d78c45000a6b1370f6d8e65da7ac52df1ae10196fd e7be859d695bdf6a19049ffb13eb1d1d2f716c742505312063ef2b2e7258b4fe 9cb7355e8b74dfb0804e72460ecebbac4931fd308f08a2ed7a9bcff2ac5768ed 7de8f721022c9bc760d2ffdae1235b0a81ece4c6fd8774712256ae29385e0246 27f8412e061a3025c472f1543acb7ce783df7472a5cfed4d079cea7a98d5bbd0 4e42fc9674517b6d3635e36d513b526cea4bbde6cfdb67616c1315b4a83286b3 07d1109c95ceb6361a6100d278e7901c258a3172f931626ec99823cf5a14927d 4640b022ee4692d2880e8f0e228ba1c0fde2015118f2cab40c004095f92410fc d14c184434948770a39005b764b8e57d78cfa76369fcd2debb9c66190c26477b 45336dfeac41982b0ce4bb72892420b3e4d9ab2ed7b4a00b66b2db6abf4159a8 612e27f8561fb6d48e72b2acdbeff5671f9ee1b7e20a133eac8bd218d27bf472 5b7c8b7b044068e1cc8f0a774e9772f9b31deb8c3536f49e70a5e5d6fc4e0a04 93b89806b7a85921780f5dd2a28bb3cc9f8fc273e23d699a82c8fc6251a5d4ca d616c20aacb0b9310d3dacc00b285b5224268abd4e8ee37b827364bab65a8f8c 5f8fd3228d034715f202c9eac40f03406ca93d97eaac09a5b72207ddd2419439 9cbfc8093367fb9ff6089283daa3b53c98980a5ce2ac88f514d090ccb4a54da6 06e080ba8c0051082f93b69a2b9dfa95f56009f74794a506e95bccf14f2303f2 c18c93713d2308c8914e9caf3ecd15b3b19090b59960bc5b5ec982a3ee5457c2 8f63380315c23754ccccb6034afdc818a22978c0cd294587a55d0c497a839e18 dd8bf31ace2648b5cb02803fe6cd112371fa0e22387fa507f07ce22d282e8b11 4bbe8eda2a48dbcd450ffa278fc1694cf712760a764bd78261e7147ef81bbdc9 5d65ad0e3bcfdc6f7bec5c2b07218e414ce1a72919a9095d04bbe7a4eccb4bad 06ac8783300828ec149a22f6a333f1edf72799e0ddb6568ca94c86a85e84fcef d67db4b0f8100a46eb0600e92aff2bca4fefaaaeeda41fab18d35b506ef8de51 ef6c2c6e815edfcdac7f8a05e42eeb71412969358ad89d4d939e7a144bf0f99b 721f7b2a84e1a4d1fe4a395fd1af2827793bab0beff766ae40cbf3a4cb6b9cf9 613e7e143552e22269da9326e4256fa4dff44fc1a6688aae7666d066248eb449 0f2da5903dfcf31d7c4421fe9dd910c94800b4401ae791b534b740b81e9fbcc6 62c879e796c92462ea2de5f164961baf41f6d222db9024495a5fe05c689e4348 22ce2fb606af969b02873ed968f10fb69f105fde801846806159acb0f4bfad2e bf93ebc964b94dd39b6a14a27bcb1224ff0fb1ce335cc441c494cedc78875dd7 cb2ccfe2110c9451827b9ac4d87f0d5273422efeeaf84e727a4ead8fbd63b04a aeddffd635950418441b864a25ff194fd04c0bc2b7dd9c93c1325afa296f5ae7 44e983dd6ee83acd0f744e5a9a5618b846c5bfe81891ac58aaf6fba78c1db8cb 925924cbf2d216b85d6c7b3a563b718a3b67293fc7d5a3a09f57659bc15de540 c640f1f5d9130a711ec3e349bd608894ff76c341730c6c7d782077d7c98bddcf 09e5e7e3f1a9f9d3cae97926ed58391f9ad577c168aecfd9ddb5acf6a9110a10 f3efffe4579ab827e5fe2f0f0dc2f8783490dfb305478186f487bb452951d6f1 f17c40af1f09a99bc6138da9fa75655267c1f22ca4b3f4ec2d03d36ee118484d c704a8185ad3cb7e97aab12c5bff71b9f5cbaeb938cb8e2adb5ac0efc946731e 773c89979714dfe22a12c8c5f30a0342f98b06bbbf5de826484f179c7624ba89 627d9a317fe3dd93172bed88d1a966876ec646a713252aa865fe35186ae3a17b]
2020-07-27T13:40:12.1536410Z --- FAIL: TestBroadcastPeers (22.79s)
2020-07-27T13:40:12.1538550Z     --- PASS: TestBroadcastPeers/OK_-_single_record (0.11s)
2020-07-27T13:40:12.1540320Z     --- PASS: TestBroadcastPeers/OK_-_single_batch_-_multiple_records (1.47s)
2020-07-27T13:40:12.1542110Z     --- FAIL: TestBroadcastPeers/OK_-_single_batch_-_max_number_of_records (2.08s)
2020-07-27T13:40:12.1543800Z     --- PASS: TestBroadcastPeers/OK_-_multiple_batches (1.62s)
2020-07-27T13:40:12.1546210Z     --- FAIL: TestBroadcastPeers/OK_-_multiple_batches_-_max_number_of_records (2.05s)
2020-07-27T13:40:12.1546690Z FAIL
2020-07-27T13:40:12.1547470Z FAIL	github.com/ethersphere/bee/pkg/hive	23.664s
2020-07-27T13:40:12.1548080Z ?   	github.com/ethersphere/bee/pkg/hive/pb	[no test files]
```

--- a/pkg/hive/hive_test.go
+++ b/pkg/hive/hive_test.go
@@ -10,8 +10,6 @@ import (
 	"fmt"
 	"io/ioutil"
 	"math/rand"
-	"reflect"
-	"sort"
 	"strconv"
 	"testing"
 	"time"
@@ -175,65 +173,78 @@ func TestBroadcastPeers(t *testing.T) {
 }
 
 func expectOverlaysEventually(t *testing.T, exporter ab.Interface, wantOverlays []swarm.Address) {
+	var (
+		overlays []swarm.Address
+		err      error
+		isIn     = func(a swarm.Address, addrs []swarm.Address) bool {
+			for _, v := range addrs {
+				if a.Equal(v) {
+					return true
+				}
+			}
+			return false
+		}
+	)
+
 	for i := 0; i < 100; i++ {
-		var stringOverlays []string
-		var stringWantOverlays []string
-		o, err := exporter.Overlays()
+		time.Sleep(50 * time.Millisecond)
+		overlays, err = exporter.Overlays()
 		if err != nil {
 			t.Fatal(err)
 		}
-		for _, k := range o {
-			stringOverlays = append(stringOverlays, k.String())
-		}
 
-		for _, k := range wantOverlays {
-			stringWantOverlays = append(stringWantOverlays, k.String())
+		if len(overlays) == len(wantOverlays) {
+			break
 		}
-
-		sort.Strings(stringOverlays)
-		sort.Strings(stringWantOverlays)
-		if reflect.DeepEqual(stringOverlays, stringWantOverlays) {
-			return
-		}
-
-		time.Sleep(10 * time.Millisecond)
 	}
 
-	o, err := exporter.Overlays()
-	if err != nil {
-		t.Fatal(err)
+	for _, v := range wantOverlays {
+		if !isIn(v, overlays) {
+			t.Errorf("overlay %s expected but not found", v.String())
+		}
 	}
 
-	t.Errorf("Overlays got %v, want %v", o, wantOverlays)
+	if t.Failed() {
+		t.Errorf("overlays got %v, want %v", overlays, wantOverlays)
+	}
 }
 
 func expectBzzAddresessEventually(t *testing.T, exporter ab.Interface, wantBzzAddresses []bzz.Address) {
+	var (
+		addresses []bzz.Address
+		err       error
+
+		isIn = func(a bzz.Address, addrs []bzz.Address) bool {
+			for _, v := range addrs {
+				if a.Equal(&v) {
+					return true
+				}
+			}
+			return false
+		}
+	)
+
 	for i := 0; i < 100; i++ {
 		time.Sleep(50 * time.Millisecond)
-		addresses, err := exporter.Addresses()
+		addresses, err = exporter.Addresses()
 		if err != nil {
 			t.Fatal(err)
 		}
 
-		if len(addresses) != len(wantBzzAddresses) {
-			continue
+		if len(addresses) == len(wantBzzAddresses) {
+			break
 		}
-
-		for i, v := range addresses {
-			if !v.Equal(&wantBzzAddresses[i]) {
-				continue
-			}
-		}
-
-		return
 	}
 
-	m, err := exporter.Addresses()
-	if err != nil {
-		t.Fatal(err)
+	for _, v := range wantBzzAddresses {
+		if !isIn(v, addresses) {
+			t.Errorf("address %s expected but not found", v.Overlay.String())
+		}
 	}
 
-	t.Errorf("Bzz addresses got %v, want %v", m, wantBzzAddresses)
+	if t.Failed() {
+		t.Errorf("bzz addresses got %v, want %v", addresses, wantBzzAddresses)
+	}
 }
 
 func readAndAssertPeersMsgs(in []byte, expectedLen int) ([]pb.Peers, error) {

--- a/pkg/hive/hive_test.go
+++ b/pkg/hive/hive_test.go
@@ -10,6 +10,7 @@ import (
 	"fmt"
 	"io/ioutil"
 	"math/rand"
+	"runtime/debug"
 	"strconv"
 	"testing"
 	"time"
@@ -202,6 +203,10 @@ func expectOverlaysEventually(t *testing.T, exporter ab.Interface, wantOverlays 
 			break
 		}
 	}
+	if len(overlays) != len(wantOverlays) {
+		debug.PrintStack()
+		t.Fatal("timed out waiting for overlays")
+	}
 
 	for _, v := range wantOverlays {
 		if !isIn(v, overlays) {
@@ -239,6 +244,10 @@ func expectBzzAddresessEventually(t *testing.T, exporter ab.Interface, wantBzzAd
 		if len(addresses) == len(wantBzzAddresses) {
 			break
 		}
+	}
+	if len(addresses) != len(wantBzzAddresses) {
+		debug.PrintStack()
+		t.Fatal("timed out waiting for bzz addresses")
 	}
 
 	for _, v := range wantBzzAddresses {

--- a/pkg/hive/hive_test.go
+++ b/pkg/hive/hive_test.go
@@ -70,7 +70,11 @@ func TestBroadcastPeers(t *testing.T) {
 		if err != nil {
 			t.Fatal(err)
 		}
-		wantMsgs[i/hive.MaxBatchSize].Peers = append(wantMsgs[i/hive.MaxBatchSize].Peers, &pb.BzzAddress{Overlay: bzzAddresses[i].Overlay.Bytes(), Underlay: bzzAddresses[i].Underlay.Bytes(), Signature: bzzAddresses[i].Signature})
+		wantMsgs[i/hive.MaxBatchSize].Peers = append(wantMsgs[i/hive.MaxBatchSize].Peers, &pb.BzzAddress{
+			Overlay:   bzzAddresses[i].Overlay.Bytes(),
+			Underlay:  bzzAddresses[i].Underlay.Bytes(),
+			Signature: bzzAddresses[i].Signature,
+		})
 	}
 
 	testCases := map[string]struct {

--- a/pkg/hive/hive_test.go
+++ b/pkg/hive/hive_test.go
@@ -70,6 +70,7 @@ func TestBroadcastPeers(t *testing.T) {
 		if err != nil {
 			t.Fatal(err)
 		}
+
 		wantMsgs[i/hive.MaxBatchSize].Peers = append(wantMsgs[i/hive.MaxBatchSize].Peers, &pb.BzzAddress{
 			Overlay:   bzzAddresses[i].Overlay.Bytes(),
 			Underlay:  bzzAddresses[i].Underlay.Bytes(),

